File: eadcrm-rest-api-examples-main/snippets/python/batch.py
"""
eAD-CRM REST API — Batch examples (Python / requests)
Module: https://www.eadvertise.eu/

Run up to 50 operations in a single request. The batch endpoint accepts
`json=` (application/json) and reuses the same tool names as the MCP server.

Install dependency:  pip install requests
"""
import requests

BASE = "https://yourdomain.com/api"
TOKEN = "YOUR_API_TOKEN"
HEADERS = {"authtoken": TOKEN}


def run_batch():
    body = {
        "operations": [
            {"tool": "customers_create", "args": {"company": "Acme LTD"}},
            {"tool": "invoices_get", "args": {"id": 1}},
        ]
    }
    r = requests.post(f"{BASE}/batch", headers=HEADERS, json=body, timeout=30)
    return r.status_code, r.json()


if __name__ == "__main__":
    print(run_batch())

File: eadcrm-rest-api-examples-main/snippets/php/batch.php
<?php
/**
 * eAD-CRM REST API — Batch examples (PHP / cURL)
 * Module: https://www.eadvertise.eu/
 *
 * Batch runs up to 50 operations in one request using the same tool names
 * as MCP. It POSTs a JSON body (Content-Type: application/json).
 *
 * No external dependencies — uses the built-in cURL extension.
 */

const BASE  = 'https://yourdomain.com/api';
const TOKEN = 'YOUR_API_TOKEN';

/** JSON request helper (sends a JSON body). */
function eadcrm_request_json(string $method, string $path, array $payload = []): array
{
    $ch = curl_init(BASE . $path);
    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_CUSTOMREQUEST  => $method,
        CURLOPT_HTTPHEADER     => [
            'authtoken: ' . TOKEN,
            'Content-Type: application/json',
        ],
        CURLOPT_POSTFIELDS     => json_encode($payload),
    ]);
    $body   = curl_exec($ch);
    $status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    curl_close($ch);

    return ['status' => $status, 'data' => json_decode((string) $body, true)];
}

// Run several operations in a single round-trip
print_r(eadcrm_request_json('POST', '/batch', [
    'operations' => [
        ['tool' => 'customers_create', 'args' => ['company' => 'Acme LTD']],
        ['tool' => 'invoices_get',     'args' => ['id' => 1]],
    ],
]));
